Extensions/Optimizely/Src/Tasks/OptimizelyX/operations/TaskOperation.ts

```diff
@@ -0,0 +1,196 @@
+      }
+
+      let holdBack: number = 10000 - (totalTraffic * 100);
+      experiment.holdback = holdBack;
+      let trafficByVariationKeyInput = this.taskInputs.getTrafficByVariation();
+      let trafficVariations = this.parseTrafficByVariation(trafficByVariationKeyInput);
+
+      for (let i: number = 0; i < experiment.variations.length; i++) {
+          let variationKey: string = (this.taskInputs.isWebProject()) ? experiment.variations[i].name : experiment.variations[i].key;
+          variationKey = variationKey.toLowerCase();
+          if (!trafficVariations.has(variationKey)) {
+              throw tl.loc("TrafficVariationKeyNotDefined", variationKey);
+          }
+          experiment.variations[i].weight = trafficVariations.get(variationKey);
+      }
+
+      return experiment;
+  }
+
+  public async getAudienceCondition(projectId: string): Promise<string> {
+      let audienceInput: string = this.taskInputs.getAudience();
+      let audience: string = null;
+
+      if (audienceInput == "" || audienceInput == null) {
+          audience = "everyone";
+      } else {
+          let result: Array<any> = new Array<any>();
+          result.push("or");
+          let audienceINputArray: string[] = audienceInput.split(',');
+          for (let i: number = 0; i < audienceINputArray.length; i++) {
+              let audienceName: string = audienceINputArray[i].trim();
+              let audienceId: number = await this.optimizelyClient.getAudienceId(projectId, audienceName);
+              result.push({"audience_id": audienceId});
+          }
+          audience = JSON.stringify(result);
+      }
+
+      console.log(tl.loc("AudienceInfo", audience));
+      return audience;
+  }
+
+  public parseTrafficByVariation(trafficByVariationString: string): Map<string, number> {
+      let arr = new Array<string>();
+      let map = new Map<string, number>();
+
+      trafficByVariationString.split('-').forEach((val) => {
+          if (!!val) {
+             arr.push(val);
+          }
+      });
+
+      for (let i: number = 0; i < arr.length; i++) {
+         let argPair: string = arr[i].trim();
+         let lastSpaceIndex: number = argPair.lastIndexOf(' ');
+
+         if (lastSpaceIndex == -1) {
+           throw tl.loc("UnableToFindValidArguments", trafficByVariationString);
+         }
+
+         let key: string = argPair.substring(0, lastSpaceIndex).trim().toLowerCase();
+         let valString: string = argPair.substring(lastSpaceIndex+1, argPair.length).trim();
+         let val: number = Number.parseInt(valString);
+
+         if (Number.isNaN(val)) {
+             throw tl.loc("NotValidValueForVariationKey", valString, key);
+         }
+
+         // we need to multiply the traffic value by 100 because the api accepts a traffic value from 0 to 10000
+         // with every 1 percent corresponding to 100 basis points. e.g. in order to vary traffic by 2 percentage points,
```

Comment:
How does Optimizely take this input in their UI? Do they also multiple by 100 or do they expect user to specify accordingly?

Reply:
they expect the user to specify this value in percentages upto 2 decimal points which when multiplied by 100 becomes a natural number. i.e. in the UI I will give a value of 23.44. But the api will send 2344